OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -147,28 +188,47 @@ public override Activity Tick(Actor self)
 				}
 			}
 
+			var exit = dest.FirstExitOrDefault(null);
+			var offset = exit != null ? exit.Info.SpawnOffset : WVec.Zero;
+
 			List<Activity> landingProcedures = new List<Activity>();
 
-			var turnRadius = Fly.CalculateTurnRadius(aircraft.Info.Speed, aircraft.Info.TurnSpeed);
+			if (aircraft.Info.CanHover)
+				landingProcedures.Add(new HeliFly(self, Target.FromPos(dest.CenterPosition + offset)));
+			else
+			{
+				var turnRadius = Fly.CalculateTurnRadius(aircraft.Info.Speed, aircraft.Info.TurnSpeed);
 
-			landingProcedures.Add(new Fly(self, Target.FromPos(w1), WDist.Zero, new WDist(turnRadius * 3)));
-			landingProcedures.Add(new Fly(self, Target.FromPos(w2)));
+				landingProcedures.Add(new Fly(self, Target.FromPos(w1), WDist.Zero, new WDist(turnRadius * 3)));
+				landingProcedures.Add(new Fly(self, Target.FromPos(w2)));
 
-			// Fix a problem when the airplane is send to resupply near the airport
-			landingProcedures.Add(new Fly(self, Target.FromPos(w3), WDist.Zero, new WDist(turnRadius / 2)));
+				// Fix a problem when the airplane is sent to resupply near the airport
```

Comment:
Line 200 - 206: In the case when `VTOL == true && CanHover == false`, e.g. Harriers, the plane should just approach the airfield in a straight line instead of circling for approach.

Reply:
Added an `else if` for that case.